gitpod-io/gitpod
# [JetBrains] Migrate backend-plugin platform version for `2024.2`

## Description


Use new beta [IntelliJ Platform Plugin](https://github.com/JetBrains/gradle-intellij-plugin) which is [required](https://plugins.jetbrains.com/docs/intellij/creating-plugin-project.html#creating-a-plugin-with-new-project-wizard) by JetBrains IDEs version `2024.2.*`

<img width="760" alt="image" src="https://github.com/gitpod-io/gitpod/assets/20944364/53ea4d88-0976-41f5-ab96-257066428aa4">


## Related Issue(s)

Fixes ENT-350, ENT-330

## How to test

- Try open workspace with stable and latest IntelliJ IDEA with repo https://github.com/mustard-mh/spring-petclinic
- For stable IDEA, everything should work like before
- For latest IDEA, everything should work like before **except** *tasks terminals*

<details>

<summary>(nit) Port forwarding</summary>

- Open workspace with latest IntelliJ IDEA with repo https://github.com/mustard-mh/spring-petclinic
- Exec command below in JetBrains terminal
```
curl lama.sh | LAMA_PORT=8000 sh
```
- It should show a port forwarding banner on top of the terminal with port-forwarding, click on it to test built-in functionalities are working well
- In `Backend Status Details` page (click right top Gitpod icon) > `Ports` tab, port `8080` should display correct state and functions like open in browser should work

</details>




## Documentation


#### Preview status

<p>Gitpod was successfully deployed to your preview environment.</p>
<ul>
	<li><b>🏷️ Name</b> - hw-eap-mig</li>
	<li><b>🔗 URL</b> - <a href="https://hw-eap-mig.preview.gitpod-dev.com/workspaces" target="_blank">hw-eap-mig.preview.gitpod-dev.com/workspaces</a>.</li>
	<li><b>📚 Documentation</b> - See our <a href="https://www.notion.so/gitpod/6debd359591b43688b52f76329d04010#7c1ce80ab31a41e29eff2735e38eec39" target="_blank">internal documentation</a> for information on how to interact with your preview environment.</li>
	<li><b>📦 Version</b> - hw-eap-mig-gha.27071</li>
	<li><b>🗒️ Logs</b> - <a href="https://console.cloud.google.com/logs/query;query=jsonPayload.kubernetes.host%3D%22preview-hw-eap-mig%22%0A%0A--%20Filter%20on%20service:%0A--%20jsonPayload.serviceContext.service%3D%22ws-manager-mk2%22%0A;duration=P1D?project=gitpod-dev-preview" target="_blank">GCP Logs Explorer</a></li>
</ul>

## Build Options

<details>
<summary>Build</summary>

      Run the build with werft instead of GHA
      Run Leeway with `--dont-test`
</details>

<details>
<summary>Publish</summary>

</details>

<details>
<summary>Installer</summary>

  Add desired feature flags to the end of the line above, space separated
</details>

<details>
<summary>Preview Environment / Integration Tests</summary>

      If enabled this will build `install/preview`
      If enabled this will create the environment on GCE infra
      Saves cost. Untick this only if you're really sure you need a non-preemtible machine.
      Valid options are `all`, `workspace`, `webapp`, `ide`, `jetbrains`, `vscode`, `ssh`. If enabled, `with-preview` and `with-large-vm` will be enabled.
</details>

/hold
